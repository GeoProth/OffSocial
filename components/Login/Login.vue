<template>
  <div class="login-form">
    <h1>Login</h1>
    <Message
      v-if="error.hasOwnProperty('message')"
      :message="error.message"
      class="error-message"
      type="error"
    />
    <Input
      v-model="username"
      class="username-input"
      holder="username"
      @enter="login({ username, password })"
    />
    <p />
    <Input
      v-model="password"
      holder="password"
      type="password"
      @enter="login({ username, password })"
    />
    <nuxt-link to="/new-user" class="new-user-link">
      <p>Don't have an account?</p>
    </nuxt-link>
    <nuxt-link to="/reset-password" class="reset-password-link">
      <p>Forgot your password?</p>
    </nuxt-link>
    <button
      text="Submit"
      class="login-form-submit"
      @click="login({ username, password })"
    >
      Submit
    </button>
  </div>
</template>
<script>
/* eslint-disable */
export default {
  name: 'Login',
  data: () => ({
    username: '',
    password: '',
    error: {}
  }),
  methods: {
    async login(creds) {
      try {
        await this.$store.dispatch('user/login', creds);
        this.$router.push('/dashboard');
      } catch (e) {
        this.error = { message: "We couldn't log you in" };
      }
    }
  }
};
</script>
<style lang="stylus" scoped>
.login-form {
  h1 {
    padding-top: 2rem
  }
  p {
    margin: 0
  }
  button {
      outline:none
      color: white;
      display: inline-block
      text-align: center
      vertical-align: middle
      border-radius: 0.5rem
      padding: 0.5rem 3rem
    :hover {
      cursor: pointer
    }
    :active {
      outline: none
      border: none
    }
    :focus { outline: 0 }
  }
}
.login-form-submit {
  margin-top: 1.5rem
  background-color: ui_blue
  color: ui_white
  font-size:15px
  padding: 5px 5px
}
.new-user-link, .reset-password-link {
  color: #000
  text-decoration: none
  p {
    margin-top: 0.5rem
    margin-bottom: 0.5rem
    font-size: 15px
  }
}

</style>
