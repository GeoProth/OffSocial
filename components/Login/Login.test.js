import FlushPromises from "flush-promises";
import Login from "./Login";
describe("Login", () => {
  let wrapper;
  const mockLogin = jest.fn();
  beforeEach(() => {
    mockLogin.mockReset();
    wrapper = mountMock(Login, {
      store: {
        user: {
          actions: {
            login: mockLogin
          }
        }
      }
    });
  });
  test("renders", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("clicking submit calls the function", async () => {
    wrapper.find(".login-form-submit").trigger("click");

    await wrapper.vm.$nextTick();
    await FlushPromises();

    expect(mockLogin).toBeCalled();
  });
  test("When the login throws an error, surface an error to the user", async () => {
    mockLogin.mockImplementation(() => {
      throw new Error("test");
    });
    wrapper = mountMock(Login, {
      store: {
        user: {
          actions: {
            login: mockLogin
          }
        }
      }
    });
    wrapper.find(".login-form-submit").trigger("click");

    await wrapper.vm.$nextTick();
    await FlushPromises();

    expect(mockLogin).toBeCalled();
    expect(wrapper.find(".error-message").isVisible()).toBe(true);
  });
});
