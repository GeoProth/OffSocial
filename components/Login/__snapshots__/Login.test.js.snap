// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders 1`] = `
<div class="login-form">
  <h1>Login</h1>
  <!----> <input placeholder="username" class="ui-input username-input">
  <p></p> <input type="password" placeholder="password" class="ui-input">
  <nuxt-link-stub to="/new-user" class="new-user-link">
    <p>Don't have an account?</p>
  </nuxt-link-stub>
  <nuxt-link-stub to="/reset-password" class="reset-password-link">
    <p>Forgot your password?</p>
  </nuxt-link-stub> <button text="Submit" class="login-form-submit">
    Submit
  </button>
</div>
`;
